float32[] obstacles_x
float32[] obstacles_y